src/app/lib/views/seedbox.js

```diff
@@ -0,0 +1,258 @@
+                    click: function () {
+                        var text = clipboard.get('text');
+                        $('#online-input').val(text);
+                    }
+                });
+
+            menu.append(cut);
+            menu.append(copy);
+            menu.append(paste);
+
+            return menu;
+        },
+
+        onNewTorrents: function (torrent) {
+          setTimeout(() => {
+            $('.notorrents-info').hide();
+            if ($(`#${torrent.infoHash}`).length || !torrent.name) {
+              return;
+            }
+
+            let activing = App.WebTorrent.torrents.filter(item => !item.paused).length;
+            if (!torrent.paused && activing >= Settings.maxActiveTorrents) {
+              this.pauseTorrent(null, torrent.infoHash);
+              torrent.paused = true;
+            }
+
+            let lastUpdated = Date.now();
+            torrent.on('download', function (bytes) {
+              if (Date.now() - lastUpdated <= 1000) {
+                return;
+              }
+
+              $(`#download-${torrent.infoHash}`).text(' ' + formatBytes(torrent.downloadSpeed) + '/s');
+              $(`#upload-${torrent.infoHash}`).text(' ' + formatBytes(torrent.uploadSpeed) + '/s');
+              lastUpdated = Date.now();
+            });
+
+            torrent.on('upload', function (bytes) {
+              if (Date.now() - lastUpdated <= 1000) {
+                return;
+              }
+
+              $(`#download-${torrent.infoHash}`).text(' ' + formatBytes(torrent.downloadSpeed) + '/s');
+              $(`#upload-${torrent.infoHash}`).text(' ' + formatBytes(torrent.uploadSpeed) + '/s');
+              lastUpdated = Date.now();
+            });
+
+            let className = "tab-torrent";
+            if ($('.tab-torrent.active').length <= 0) {
+              className += ' active';
+            }
+
+            $('.seedbox-torrent-list>ul.file-list').append(
+              `<li class="${className}" id="${torrent.infoHash}" data-season="1" data-episode="10">
+                <a href="#" class="episodeData">
+                    <span>1</span>
+                    <div>${torrent.name}</div>
+                </a>
+
+                <i class="fa fa-trash-alt watched trash-torrent" id="trash-${torrent.infoHash}"></i>
+                <i class="fa fa-play watched resume-torrent" id="play-${torrent.infoHash}" style="display: ${torrent.paused ? '' : 'none'};"></i>
+                <i class="fa fa-pause-circle watched pause-torrent" id="resume-${torrent.infoHash}" style="display: ${torrent.paused ? 'none' : ''};"></i>
+                <i class="fa fa-upload watched" id="upload-${torrent.infoHash}"> 0 Kb/s</i>
+                <i class="fa fa-download watched" id="download-${torrent.infoHash}"> 0 Kb/s</i>
+              </li>`
+            );
+
+            $('.seedbox-overview').show();
+          }, 500 + Math.floor(Math.random() * Math.floor(500)));
```

Comment:
What the heck is this here for? I'm going through the code for an unrelated change and this just seems wholly unnecessary.

Reply:
@ArcticZeroo 500+ milliseconds is necessary to make sure the torrent is fully initialized, as you can see in the first line for this function, I also check `if ($(`#${torrent.infoHash}`).length || !torrent.name)`. So what will happen if no delay in the function? => nothing will be add to the seedbox UI.

Reply:
@ArcticZeroo I'm not sure why I added the setTimeout here, but it must be for a reason. You're welcome to test, remove it and create a new merger request. Thank you.

Reply:
The solution to an object that takes time to initialize is not to wait 500ms + a random amount of milliseconds (seriously, how does that not seem crazy?), but rather to wait for it to be ready, e.g. `torrent.on('ready')`

Reply:
Thank, it's my bad. i wrote the code for my daily use only - and didn't spend more time to optimize, then I realized It's useful and make a new merge request without any change. Please help to improve it.

Reply:
it probably needs more cleanup eventually, but this particular change is in https://github.com/popcorn-official/popcorn-desktop/pull/1560